docs/harvester-configuration.md

```diff
@@ -0,0 +1,281 @@
+# Harvester Configuration
+
+Harvester configuration file can be provided during manual or automatic installation to configure various settings. The following is an configuration example:
+
+```yaml
+server_url: https://someserver:6443
+token: TOKEN_VALUE
+os:
+  ssh_authorized_keys:
+  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQAB...
+  - github:username
+  hostname: myhost
+  modules:
+  - kvm
+  - nvme
+  sysctl:
+    kernel.printk: "4 4 1 7"
+    kernel.kptr_restrict: "1"
+  dns_nameservers:
+  - 8.8.8.8
+  - 1.1.1.1
+  ntp_servers:
+  - 0.us.pool.ntp.org
+  - 1.us.pool.ntp.org
+  wifi:
+  - name: home
+    passphrase: mypassword
+  - name: nothome
+    passphrase: somethingelse
+  password: rancher
+  environment:
+    http_proxy: http://myserver
+    https_proxy: http://myserver
+install:
+  mode: create
+  mgmtInterface: eth0
+  force_efi: true
+  device: /dev/vda
+  silent: true
+  iso_url: http://myserver/test.iso
+  poweroff: true
+  no_format: true
+  debug: true
+  tty: ttyS0
+```
+
+## Configuration Reference
+
+The configuration file is a YAML file and derived from the [k3os configuration](https://github.com/rancher/k3os#configuration-reference). Below is a reference of all configuration keys.
```

Comment:
Should we need to mention k3os? because actually your implementation already has the implicit mapping between harvester config and k3os, but I am fine with it if you have reasons.

Reply:
Removed.